packages/jest-emotion/src/matchers.js

```diff
@@ -35,13 +49,35 @@ function valueMatches(declaration, value) {
   return value === declaration.value
 }
 
-function toHaveStyleRule(received: *, property: *, value: *) {
+function toHaveStyleRule(
+  received: *,
+  property: *,
+  value: *,
+  options?: { target?: string, media?: string } = {}
+) {
+  const { target = '', media = '' } = options
   const classNames = getClassNamesFromNodes([received])
   const cssString = getStylesFromClassNames(classNames, getStyleElements())
   const styles = css.parse(cssString)
 
-  const declaration = styles.stylesheet.rules
-    .reduce((decs, rule) => Object.assign([], decs, rule.declarations), [])
+  let preparedRules = styles.stylesheet.rules
+  if (media.length > 1) {
+    if (shortMediaRegExp.test(media) || longMediaRegExp.test(media)) {
```

Comment:
Will `longMediaRegExp` break when more than one `and` clause is preset?

`@media screen and (min-width: 30em) and (orientation: landscape)`

Reply:
Yea, it will break I will fix this.

Reply:
Just checked this scenario and it is passed. But I think we can get rid of `longMediaRegExp` because it is don't extend `shortMediaRegExp` properly and a invalid `media` can pass this check by adding `() and ()` pattern